zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreterServer.java

```diff
@@ -140,10 +136,9 @@ public static void main(String[] args)
 
 
   @Override
-  public void createInterpreter(String interpreterGroupId, String className, Map<String, String>
-          properties)
-      throws TException {
-
+  public void createInterpreter(String interpreterGroupId, String noteId, String
```

Comment:
open() / close() can be invoked even without notebook creation. but InterpreterContext has some notebook / paragraph related values. except for that i don't see big problem for now (didn't have much time to have think thoughtfully).

I was also thinking make `Interpreter.completion()` return `List<Completion>` instead of `List<String>`. so completion can show more informations. not only method name, but also parameters, etc.

How about start a new discussion for updating Interpreter api?